lua/ttt2/libraries/outline.lua

```diff
@@ -20,37 +25,172 @@ OUTLINE_MODE_BOTH = 0
 OUTLINE_MODE_NOTVISIBLE = 1
 OUTLINE_MODE_VISIBLE = 2
 
-local List, ListSize = {}, 0
+OUTLINE_RENDERTYPE_BEFORE_VM = 0	-- Render before drawing the view model
+OUTLINE_RENDERTYPE_BEFORE_EF = 1	-- Render before drawing all effects (after drawing the viewmodel)
+OUTLINE_RENDERTYPE_AFTER_EF = 2		-- Render after drawing all effects
+
+local Lists = {
+    [OUTLINE_RENDERTYPE_BEFORE_VM] = {},
+    [OUTLINE_RENDERTYPE_BEFORE_EF] = {},
+    [OUTLINE_RENDERTYPE_AFTER_EF] = {}
+}
+
+local ListsSize = {
+    [OUTLINE_RENDERTYPE_BEFORE_VM] = {},
+    [OUTLINE_RENDERTYPE_BEFORE_EF] = {},
+    [OUTLINE_RENDERTYPE_AFTER_EF] = {}
+}
 local RenderEnt = NULL
+local RenderType = OUTLINE_RENDERTYPE_AFTER_EF
+local OutlineThickness = 1
+local texture_store = render.GetScreenEffectTexture(0)
+local texture_draw = render.GetScreenEffectTexture(1)
 
-local outlineMatSettings = {
+local outline_mat_settings = {
+    ["$basetexture"] = texture_draw:GetName(),
     ["$ignorez"] = 1,
     ["$alphatest"] = 1,
+    ["$smooth"] = 0
 }
+local outline_mat = CreateMaterial("ttt2_outline", "UnlitGeneric", outline_mat_settings)
 
-local copyMat = Material("pp/copy")
-local outlineMat = CreateMaterial("OutlineMat", "UnlitGeneric", outlineMatSettings)
-local storeTexture = render.GetScreenEffectTexture(0)
-local drawTexture = render.GetScreenEffectTexture(1)
+local copy_mat = Material("pp/copy")
 
 local ENTS = 1
 local COLOR = 2
 local MODE = 3
 
+--FIXME: You can decide if you want this to be a option for users, should be adressed before merge
+local enable_thin_line_workaround = false
```

Comment:
we do have a performance section in the appearance menu. You could add it there if you like, enabled by default probably

Reply:
It really hinges on whether you find the appearance of them being slighty thicker fine, its super subjective but it does look better than the thin line scenarios where it can get cutoff

Reply:
you know what, if the performance is mostly similar, chose whatever you believe looks best and set it as default

Reply:
no perf difference, it just adjusts the offset by 1 pixel, theres 1 edge case where this looks worse and thats if you have a 1 pixel thick line of geometry on a model being outlined but thats exceptionally rare and even then you'd have to zoom in to notice it, the problem with lines being too thin is a lot more common and noticeable, i'll leave what looks best to you guys in this case because my opinion keeps flipping dependant on which ive been looking at longer so im kinda neutral on it

Reply:
to me your discription sounds like it should be enabled